modules/photo/src/merge.cpp

```diff
@@ -173,93 +173,99 @@ class MergeMertensImpl CV_FINAL : public MergeMertens
         std::vector<Mat> weights(images.size());
         Mat weight_sum = Mat::zeros(size, CV_32F);
 
-        for(size_t i = 0; i < images.size(); i++) {
-            Mat img, gray, contrast, saturation, wellexp;
-            std::vector<Mat> splitted(channels);
-
-            images[i].convertTo(img, CV_32F, 1.0f/255.0f);
-            if(channels == 3) {
-                cvtColor(img, gray, COLOR_RGB2GRAY);
-            } else {
-                img.copyTo(gray);
-            }
-            split(img, splitted);
+        parallel_for_(Range(0, static_cast<int>(images.size())), [&](const Range& range) {
+            for(int i = range.start; i < range.end; i++) {
+                Mat gray, contrast, saturation, wellexp;
+                std::vector<Mat> splitted(channels);
 
-            Laplacian(gray, contrast, CV_32F);
-            contrast = abs(contrast);
+                images[i].convertTo(images[i], CV_32F, 1.0f/255.0f);
```

Comment:
Could we keep using `img` instead of multiple `images[i]`?

With declaring it in this way:

```
Mat& img = images[i];
```

Reply:
Sure, done.